framework/src/utils/MathUtils.C

```diff
@@ -202,6 +204,114 @@ multiIndex(unsigned int dim, unsigned int order)
   return multi_index;
 }
 
+Point
+barycentricToCartesian2D(const Point & p0,
```

Comment:
why "to Cartesian"?

Reply:
Because it converts from barycentric _to Cartesian_ coordinates.

Reply:
wow